tests/EventStoreConnectionTestCase.php

```diff
@@ -28,8 +29,10 @@ protected function setUpAsync(): Generator
         yield $this->connection->connectAsync();
     }
 
-    protected function tearDownAsync()
+    protected function tearDownAsync(): Generator
     {
         $this->connection->close();
+
+        yield new Success();
```

Comment:
why is this necessary? :thinking:

Reply:
Ah I think I understand now.

Reply:
Because then it's not a generator. tearDown (non-async) is called later, so the test would hang otherwise forever (unless you get to unreference watchers).

Reply:
If unreferencing the connection works, then the method can be deleted here anyway, as we don't need to close the connection anymore.